File: backend/KSWeb.Api/Models/DashboardModels.cs
namespace KSWeb.Api.Models;

public sealed record DashboardResumo(
    int TotalOS,
    int TotalAbertas,
    int TotalComResolucao,
    int TotalSemResolucao,
    IReadOnlyList<DashboardStatusItem> Status);

public sealed class DashboardStatusItem
{
    public long StatusId { get; set; }

    public string Status { get; set; } = string.Empty;

    public long Total { get; set; }
}

File: frontend/ksweb-app/src/app/pages/dashboard/dashboard.models.ts
export interface DashboardStatusItem {
  statusId: number;
  status: string;
  total: number;
}

export interface DashboardResumo {
  totalOS: number;
  totalAbertas: number;
  totalComResolucao: number;
  totalSemResolucao: number;
  status: DashboardStatusItem[];
}
